src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodToStateMachineRewriter.cs

```diff
@@ -116,6 +116,22 @@ internal void GenerateMoveNext(BoundStatement body, MethodSymbol moveNextMethod)
         {
             F.CurrentMethod = moveNextMethod;
 
+            BoundStatement newBody;
+            try
+            {
+                newBody = GenerateMoveNextBody(body);
+            }
+            catch (SyntheticBoundNodeFactory.MissingPredefinedMember ex)
+            {
+                F.Diagnostics.Add(ex.Diagnostic);
+                newBody = new BoundBadStatement(body.Syntax, ImmutableArray.Create<BoundNode>(body), hasErrors: true);
+            }
+
+            F.CloseMethod(newBody);
```

Comment:
Will leave in the `AsyncRewriter` since this matches the `try/catch` in `LocalRewriter`.